plugins/scaffolder/src/api.ts

```diff
@@ -153,8 +158,9 @@ export class ScaffolderClient implements ScaffolderApi {
   async scaffold(
     templateName: string,
     values: Record<string, any>,
+    githubAccessToken: string,
   ): Promise<string> {
-    const token = await this.identityApi.getIdToken();
+    const token = githubAccessToken ?? (await this.identityApi.getIdToken());
```

Comment:
Also don't think that we should be overriding the token here either, we still need the backstage ID token if it's set, thinking that this auth part should be something seperate sent alongside the values so that we can write it somewhere seperate to the values that's not going to be surfaced by the API, but accessible to the action